### PR TITLE
refactor(api): change splits and segments to maps with keys that match data loader expectations

### PR DESCRIPTION
**Pull request by celiawaggoner**
next step: update the data serializer to stringify individual split and segment objects

### CODE REVIEW

File: serializer/serializer.go [hunk 1]
```diff
@@ -22,14 +29,20 @@ func NewSerializer(poller poller.Fetcher) *Serializer {
...
-	if reflect.DeepEqual(latestData, poller.SplitData{}) {
-		return fmt.Sprintf(formattedLoggingScript, "{}"), nil
-	}
-
-	splitCachePreload, err := json.Marshal(latestData)
-	if err != nil {
-		return "", err
-	}
-	return fmt.Sprintf(formattedLoggingScript, string(splitCachePreload)), nil
+        latestData := serializer.poller.GetSplitData()
```
Comment (rohanmurthy) [inline serializer/serializer.go]: nit: I believe the tab formatting may be off

Comment (celiawaggoner) [reply serializer/serializer.go]: whoops, yep -- ty!

File: api/api_test.go [hunk 2]
```diff
@@ -411,8 +413,10 @@ func TestGetSegmentsForSplitsReturnsValid(t *testing.T) {
 	defer testServer.Close()
 	conditions := []dtos.ConditionDTO{}
 	json.Unmarshal([]byte(mockConditions), &conditions)
-	split := dtos.SplitDTO{Conditions: conditions}
-	splits := []dtos.SplitDTO{split}
+        split := dtos.SplitDTO{Name: "mock-split", Conditions: conditions}
+        splits := map[string]dtos.SplitDTO{
```
Comment (wendychiang) [inline api/api_test.go]: these are also tabs

File: serializer/serializer_test.go [hunk 3]
```diff
@@ -87,8 +86,7 @@ func TestGetSerializedDataMarshalEmptyCache(t *testing.T) {
 	result, err := serializer.GetSerializedData()
 
 	// Validate that returned logging script contains a valid SplitData
-	expectedData := "{}"
-	expectedLoggingScript := fmt.Sprintf(formattedLoggingScript, expectedData)
+        expectedLoggingScript := "\n<script>\n  window.__splitCachePreload = {\n  splitsData: null,\n  since: 0,\n  segmentsData: null,\n  usingSegmentsCount: 0\n  };\n</script>"
```
Comment (wendychiang) [inline serializer/serializer_test.go]: I think node version made a change to return "{}" in this case and we want to do the same thing for Go?
https://github.com/godaddy/split-node-serializer/pull/27

Comment (celiawaggoner) [reply serializer/serializer_test.go]: yup, i'll do that in a follow up pr

**Pull request merged**